backend/src/org/commcare/suite/model/Entry.java

```diff
@@ -19,23 +19,15 @@
 import java.util.Vector;
 
 /**
- * <p>An Entry definition describes a user
- * initiated form entry action, what information
- * needs to be collected before that action can
- * begin, and what the User Interface should
- * present to the user regarding these actions</p>
+ * Describes a user-initiated form entry action. Includes information that
+ * needs to be collected before that action can begin and what the UI should
+ * present to the user regarding this action.
  *
  * @author ctsims
  */
-public class Entry implements Externalizable, MenuDisplayable {
+public class Entry extends EntryBase {
```

Comment:
could consider renaming this class to FormEntry, and then you could call EntryBase just Entry